src/rard/research/migrations/0061_order_in_book.py

```diff
@@ -0,0 +1,108 @@
+# Generated by Django 3.2 on 2023-01-23 11:20
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("research", "0060_add_reference_model"),
+    ]
+
+    def create_unknown_work(apps, schema_editor):
+        Antiquarian = apps.get_model("research", "Antiquarian")
+        antiquarians = Antiquarian.objects.all()
+        for ant in antiquarians:
+            ant.add_work(name="Unknown Work", unknown=True)
+
+    def create_unknown_book(apps, schema_editor):
+        Work = apps.get_model("research", "Work")
+        works = Work.objects.all()
+        for work in works:
+            work.add_book(name="Unknown Book", unknown=True)
+
+    def assign_links_to_unknown(apps, schema_editor):
+        worklinks = ["FragmentLink", "TestimoniumLink", "AppositumFragmentLink"]
+
+        for link_type in worklinks:
+            links = link_type.object.all()
+            for link in links:
+                ant = link.antiquarian
+                if link.work == None:
+                    link.work = ant.works.get(unknown=True)
+                work = link.work
+                if link.book == None:
+                    link.book = work.books.get(unknown=True)
+
+    def give_books_order(apps, schema_editor):
+        Work = apps.get_model("research", "Work")
+        for work in Work.objects.all():
+            books = work.book_set.all().order_by("number")
+            for count, book in enumerate(books):
+                book.order = count
+                book.save()
+
+    def give_links_order_in_book(apps, schema_editor):
+        worklinks = ["FragmentLink", "TestimoniumLink", "AppositumFragmentLink"]
+
+        Work = apps.get_model("research", "Work")
+
+        def order_in_book_for_link_class(link_classname):
+            link_class = apps.get_model("research", link_classname)
+            for work in Work.objects.all():
+                related_links = link_class.objects.filter(work=work).order_by(
+                    "book__order", "work_order"
+                )
+                for count, link in enumerate(related_links):
+                    link.order_in_book = count
+                    link.save()
```

Comment:
I think this isn't quite right. We want order_in_book to be 0 to n-1 for the n links associated with each Book, but this would set order_in_book to 0 to m-1 for the m links associated with each Work. So, we want to loop through each Book, get related links ordered by work_order, then loop through the queryset and assign order_in_book.